api/kogito-api/src/main/java/org/kie/kogito/rules/DataObserver.java

```diff
@@ -16,12 +16,12 @@ public FactHandle insert(DataHandle handle, T object) {
 
             @Override
             public void update(DataHandle handle, T object) {
-
+                consumer.accept(object);
             }
 
             @Override
             public void delete(DataHandle handle) {
-
+                consumer.accept(null);
```

Comment:
I think we can accept this semantics for now, but in general I'd rather create another kind of DataObserver which "wraps" the type of event (`consumer.accept(new Updated(object))`), or i would define a specific DataObserver for this special purpose